api/layer/object_cache.go

```diff
@@ -74,6 +86,31 @@ func (l *listObjectsCache) Put(key string, objects []*ObjectInfo) {
 	})
 }
 
-func createKey(accessKey string, cid *cid.ID) string {
-	return accessKey + cid.String()
+func createKey(ctx context.Context, cid *cid.ID) (string, error) {
+	box, err := GetBoxData(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return box.Gate.AccessKey + cid.String(), nil
+}
+
+func trimAfterObjectName(startAfter string, objects []*ObjectInfo) []*ObjectInfo {
+	if objects != nil && len(startAfter) != 0 && objects[0].Name <= startAfter {
```

Comment:
If objects is `[]*ObjectInfo{}` panic will occur.

Reply:
Fixed